### PR TITLE
Fix lint warnings in URL polyfill

### LINKED ISSUE #6872: Fix lint warnings in URL polyfill
Follow up to #6846

### PR DESCRIPTION
**Pull request by yashsriv**
Fixes #6872.

### CODE REVIEW

**Comment by yashsriv** [comment]
I have a doubt regarding the object which was called `jURL` earlier. JSHint was throwing an error regarding UpperCamelCase for Constructor names so I chose `JURL` but it doesn't look very nice.

P.S. This is my first P.R. on any open source project ever. Please suggest changes if any.

File: src/shared/util.js [hunk 1]
```diff
@@ -1999,28 +2003,33 @@ function loadJpegStream(id, imageUrl, objs) {
...
+              if (':' === cp && null === this._password) {
                 this._password = '';
                 continue;
               }
               var tempC = percentEscape(cp);
-              (null !== this._password) ? this._password += tempC : this._username += tempC;
+              if (null !== this._password) {
+                  this._password += tempC;
+              } else {
+                  this._username += tempC;
```
Comment (timvandermeij) [inline src/shared/util.js]: Remove two spaces here.

File: src/shared/util.js [hunk 2]
```diff
@@ -1999,28 +2003,33 @@ function loadJpegStream(id, imageUrl, objs) {
...
               // XXX check URL code points
-              if (':' == cp && null === this._password) {
+              if (':' === cp && null === this._password) {
                 this._password = '';
                 continue;
               }
               var tempC = percentEscape(cp);
-              (null !== this._password) ? this._password += tempC : this._username += tempC;
+              if (null !== this._password) {
+                  this._password += tempC;
```
Comment (timvandermeij) [inline src/shared/util.js]: Remove two spaces here.

File: src/shared/util.js [hunk 3]
```diff
@@ -1931,12 +1934,13 @@ function loadJpegStream(id, imageUrl, objs) {
...
           } else {
-            var nextC = input[cursor+1]
-            var nextNextC = input[cursor+2]
+            var nextC = input[cursor+1];
+            var nextNextC = input[cursor+2];
             if (
-              'file' != this._scheme || !ALPHA.test(c) ||
-              (nextC != ':' && nextC != '|') ||
-              (EOF != nextNextC && '/' != nextNextC && '\\' != nextNextC && '?' != nextNextC && '#' != nextNextC)) {
+              'file' !== this._scheme || !ALPHA.test(c) ||
```
Comment (timvandermeij) [inline src/shared/util.js]: Put this on the previous line and align the next lines under it.

File: src/shared/util.js [hunk 4]
```diff
@@ -1840,9 +1840,10 @@ function loadJpegStream(id, imageUrl, objs) {
             if (isRelativeScheme(this._scheme)) {
               this._isRelative = true;
             }
-            if ('file' == this._scheme) {
+            if ('file' === this._scheme) {
               state = 'relative';
-            } else if (this._isRelative && base && base._scheme == this._scheme) {
+            } else if (this._isRelative && base &&
+                    base._scheme === this._scheme) {
```
Comment (timvandermeij) [inline src/shared/util.js]: Align this line with `this` on the previous line.

File: src/shared/util.js [hunk 5]
```diff
@@ -1788,20 +1787,20 @@ function loadJpegStream(id, imageUrl, objs) {
...
-       [0x22, 0x23, 0x3C, 0x3E, 0x60].indexOf(unicode) == -1
+       [0x22, 0x23, 0x3C, 0x3E, 0x60].indexOf(unicode) === -1
       ) {
       return c;
     }
     return encodeURIComponent(c);
   }
 
-  var EOF = undefined,
+  var EOF,
```
Comment (timvandermeij) [inline src/shared/util.js]: Try to put these on one line if possible (or otherwise two).

File: src/shared/util.js [hunk 6]
```diff
@@ -1811,7 +1810,8 @@ function loadJpegStream(id, imageUrl, objs) {
         seenBracket = false,
         errors = [];
 
-    loop: while ((input[cursor - 1] != EOF || cursor == 0) && !this._isInvalid) {
+    loop: while ((input[cursor - 1] !== EOF || cursor === 0) &&
+                  !this._isInvalid) {
```
Comment (timvandermeij) [inline src/shared/util.js]: Remove one space here to align with the second `(` on the previous line.

**Comment by timvandermeij** [comment]
@yashsriv Thank you for your contribution! Your patch looks good and I have added a few comments to improve it. Linting does pass, so that's really nice! When you address these comments, please squash the commits into one afterwards (see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits on how to do this easily). If you have any questions, feel free to ask!

**Comment by Snuffleupagus** [comment]
Also it'd be great if the commit message was updated to e.g. `Fix lint warnings in URL polyfill` or similar, since the current `Addresses #6872 - Fixed linting for the file src/shared/util.js` doesn't really give much information about what changed and why!

**Comment by yashsriv** [comment]
Updated as per the comments.

**Comment by timvandermeij** [comment]
Nice work, thank you!

**Pull request merged**